### DISCUSSION TITLE
Long running tasks

### DISCUSSION THREAD

**Discussion by etelford** [brefphp/bref#1476] [Support]
I'm using Bref v1 and I have a task that takes longer than 28 seconds to run. I attempted to invoke the function directly instead of via the API gateway, but I still get a FPM timeout after the storm standard 28 seconds.

Is there any technique I can use that will allow for longer a timeout when my function is invoked?

Thank you.

**Comment by mnapoli** [comment]
We'll need more info to understand, did you set a timeout of 29 seconds on your function?

Also from what I gather this isn't an HTTP route? Then maybe it would be worth having a function that is not exposed via HTTP (https://bref.sh/docs/runtimes/function.html), and thus could go up to 15 minutes.

**Comment by etelford** [comment]
I ended up solving this by using a console version of my application.
